src/Sulu/Component/Content/Types/Block/ScheduleBlockVisitor.php

```diff
@@ -0,0 +1,123 @@
+use Sulu\Component\Webspace\Analyzer\RequestAnalyzerInterface;
+
+class ScheduleBlockVisitor implements BlockVisitorInterface
+{
+    /**
+     * @var RequestAnalyzerInterface
+     */
+    private $requestAnalyzer;
+
+    /**
+     * @var CacheLifetimeRequestEnhancer
+     */
+    private $cacheLifetimeRequestEnhancer;
+
+    public function __construct(
+        RequestAnalyzerInterface $requestAnalyzer,
+        CacheLifetimeRequestEnhancer $cacheLifetimeRequestEnhancer
+    ) {
+        $this->requestAnalyzer = $requestAnalyzer;
+        $this->cacheLifetimeRequestEnhancer = $cacheLifetimeRequestEnhancer;
+    }
+
+    public function visit(BlockPropertyType $block): ?BlockPropertyType
+    {
+        $blockPropertyTypeSettings = $block->getSettings();
+
+        if (!\is_array($blockPropertyTypeSettings)
+            || !isset($blockPropertyTypeSettings['schedules_enabled'])
+            || !$blockPropertyTypeSettings['schedules_enabled']
+            || !isset($blockPropertyTypeSettings['schedules'])
+            || !\is_array($blockPropertyTypeSettings['schedules'])
+        ) {
+            return $block;
+        }
+
+        $now = $this->requestAnalyzer->getDateTime();
+        $nowTimestamp = $now->getTimestamp();
+
+        $returnBlock = false;
+
+        foreach ($blockPropertyTypeSettings['schedules'] as $schedule) {
+            switch ($schedule['type']) {
+                case 'fixed':
+                    $start = new \DateTime($schedule['start']);
+                    $end = new \DateTime($schedule['end']);
+
+                    $this->cacheLifetimeRequestEnhancer->setCacheLifetime($start->getTimestamp() - $nowTimestamp);
+                    $this->cacheLifetimeRequestEnhancer->setCacheLifetime($end->getTimestamp() - $nowTimestamp);
+
+                    if ($now >= $start && $now <= $end) {
+                        $returnBlock = true;
+                        continue 2;
+                    }
+                    break;
+                case 'weekly':
+                    $year = $now->format('Y');
+                    $month = $now->format('m');
+                    $day = $now->format('d');
+
+                    $start = new \DateTime($schedule['start']);
+                    $start->setDate($year, $month, $day);
+                    $end = new \DateTime($schedule['end']);
+                    $end->setDate($year, $month, $day);
+
+                    if ($end < $start) {
+                        // If the end date is smaller than the start date, it means that the user has entered a time
+                        // combination that spans multiple days. In order to make sure that the start date is before the
+                        // end date the start date has to be set to yesterday.
+                        $start->modify('-1 day');
+                    }
+
+                    $i = 0;
+                    // This loop checks for the coming 7 days for matches of the weekdays selected by the users. This is
```

Comment:
```suggestion
                    // This loop checks the coming 7 days for matches of the weekdays selected by the user. This is
```

Reply:
The "for" is doubled this way, which is not correct.

Reply:
but the `for` is doubled right now, isnt it? 🤔

Reply:
Ah, sorry, read the diff the wrong way :see_no_evil: